packages/modules/product/src/services/product-module-service.ts

```diff
@@ -941,7 +938,7 @@ export default class ProductModuleService<
           sharedContext
         )
 
-        if (productsToUpdate.length > 0) {
+        if (productsToUpdate?.length > 0) {
```

Comment:
```suggestion
const updateSelectorAndData = updatedCollections.map((collectionData) => {
  const input = normalizedInput.find((c) => c.id === collection.id)
  const productsToUpdate = (input as any)?.products

  const dissociateSelector = {
    collection_id: collectionData.id,
  }
  const associateSelector = {}

  if (!!productsToUpdate?.length) {
    dissociateSelector["id"] = { $nin: productsToUpdate.map((p) => p.id) }
    associateSelector["id"] = { $in: productsToUpdate.map((p) => p.id) }
  }

  const result: Record<string, any>[] = [{
    selector: dissociateSelector,
    data: {
      collection_id: null,
    },
  }]
  
  if (isPresent(associateSelector)) {
    result.push({
      selector: associateSelector,
      data: {
        collection_id: collectionData.id,
      },
    })
  }
  
  return result
}).flat()

await this.productService_.update(
  updateSelectorAndData,
  sharedContext
)
```

can you tell me if this could work, so that we don't have to have a promise all of many promises of potentially unknown size